wallet/src/de/schildbach/wallet/service/WalletTransactionMetadataProvider.kt

```diff
@@ -78,15 +87,20 @@ class WalletTransactionMetadataProvider @Inject constructor(
             )
             transactionMetadataDao.insert(metadata)
             log.info("txmetadata: inserting $metadata")
+
+            return metadata
         }
+
+        return null
     }
 
-    private suspend fun updateAndInsertIfNotExist(txId: Sha256Hash, update: suspend () -> Unit) {
-        if (transactionMetadataDao.exists(txId)) {
-            update()
+    private suspend fun updateAndInsertIfNotExist(txId: Sha256Hash, update: suspend (TransactionMetadata) -> Unit) {
+        val existing = transactionMetadataDao.load(txId)
+
+        if (existing != null) {
```

Comment:
Returning `TransactionMetadata` here allows us to quickly update several fields using `.copy()` and `@Update` functions.